sqlx-macros/src/lib.rs

```diff
@@ -60,6 +62,16 @@ macro_rules! async_macro (
         let res: Result<proc_macro2::TokenStream> = block_on(async {
             use sqlx::connection::Connect;
 
+            // If a .env file exists at CARGO_MANIFEST_DIR, load environment variables from this,
+            // otherwise fallback to default dotenv behaviour.
+            if let Ok(dir) = std::env::var("CARGO_MANIFEST_DIR") {
+                let env_path = PathBuf::from(&dir).join(".env");
```

Comment:
Sorry for the confusion, I previously meant this should be `Path::from(&dir)` but you'd actually need to deref the string to a slice since there's no `impl From<&String> for Path`.

As written, this is actually copying the string twice before appending `/.env` to it; once in `PathBuf::from(&dir)` (which must copy the string) and then again in `.join()` (which clones the pathbuf before appending to it in-place and returning the updated copy).

You can eliminate the copying entirely but I don't think it's worth the extra verbosity for code that's not in a hot loop. I would just drop the `&` so it's `PathBuf::from(dir).join(".env")` so it's at least not copying the string twice.

Reply:
Aha, got it, change made!